File: backend-java/SpringCloudPXL/shoppingcart-service/src/main/java/be/pxl/services/domain/ShoppingCart.java
package be.pxl.services.domain;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Entity
@Table(name="shoppingcart")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ShoppingCart {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    private Long userId;
    private Double totalPrice;
    private Boolean checkedOut;
    @OneToMany(mappedBy = "shoppingCart", cascade = CascadeType.ALL)
    private List<ShoppingCartProduct> products;
}

File: frontend-web/e-commerceplatform/src/app/models/shoppingcart.ts
import { Product } from "./product";
import { ShoppingCartProduct } from "./shoppingcart-product";

export class Shoppingcart {
    id: number;
    userId: string;
    totalPrice: number;
    checkedOut: boolean;
    products: ShoppingCartProduct[];

    constructor(id: number, userId: string, products: ShoppingCartProduct[], totalPrice: number, checkedOut: boolean) {
        this.id = id;
        this.userId = userId;
        this.products = products;
        this.totalPrice = totalPrice;
        this.checkedOut = checkedOut;
    }
}
